packages/frontend/src/hooks/useMessageSentNotifier.ts

```diff
@@ -0,0 +1,63 @@
+import { useState, useCallback, useEffect } from 'react'
+import type { T } from '@deltachat/jsonrpc-client'
+
+// Simple global state for message notifications
+const messageListeners: Set<
+  (accountId: number, chatId: number, message: Partial<T.MessageData>) => void
+> = new Set()
+
+export function notifyMessageSent(
+  accountId: number,
+  chatId: number,
+  message: Partial<T.MessageData>
+) {
+  messageListeners.forEach(callback => callback(accountId, chatId, message))
+}
+
+/**
+ * A notifier that allows components to subscribe to message sent events.
+ * For cases where DC Event 'MsgsChanged' is not sufficient
+ * (like on MainScreen where we want to update AppIcons immediately when a
+ * Webxdc message is sent).
+ */
+export function useMessageSentNotifier() {
+  const [listeners] = useState(() => messageListeners)
```

Comment:
Why this weird `useState` thing? Why not `useRef` or, better yet, using the `messageListeners` var directly?

Reply:
Is removed. Was a left over...